### ISSUE TITLE
Pie Charts produce wrong labels on hover when some items have value of 0

### ISSUE THREAD

**Issue by dmathisen** [ag-grid/ag-grid#6420]
**I'm submitting a ...**  (check one with "x")
```
[x] bug report => see 'Providing a Reproducible Scenario'
[] feature request => do not use Github for feature requests, see 'Customers of AG Grid'
[] support request => see 'Requesting Community Support'
```

**Providing a Reproducible Scenario**
Reproducible Plunkr example: https://plnkr.co/edit/AWMUigmpqOzUBKdE?open=index.jsx

Create a pie chart and have at least 1 field with data = 0
The callout labels are wrong
In this plunkr example above, I made "Blackberry" and "Symbian" have a value of 0.
Now when I hover over the iOS or Android slices, the tooltip shows "Symbian

![image](https://user-images.githubusercontent.com/684701/229571996-9c275165-5510-4856-bcea-868a749bfafb.png)

![image](https://user-images.githubusercontent.com/684701/229571627-3ade19fa-3fc3-492c-810f-95ac0472f7f0.png)

**Please tell us about your environment:**
Windows 10, VS Code, Npm

* **AG Grid version:** 29.2.0

* **Browser:**
Edge
Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/111.0.0.0 Safari/537.36 Edg/111.0.1661.54

* **Language:** TypeScript 5.0.2

**Comment by StephenCooper** [comment]
Thanks for reporting this. I have raised an internal ticket for further investigation.

**Comment by StephenCooper** [comment]
Based on your description we've now reproduced this issue as a bug.
 
We have added this bug to our backlog and we are tracking it with the following reference and description:
AG-86321 [Charts] Pie chart not behaving correctly when some items have value of 0
 
We try to fix bugs from one release to the next, so this should be fixed in the next release or the one after if it was raised too close to the next release date.
 
See whether this item will be in the next release by checking the NEXT RELEASE checkbox on the product pipeline page:
https://www.ag-grid.com/ag-grid-pipeline/
 
The best way to track this is to sign up for AG Grid new release notifications using [the instructions here](https://ag-grid.zendesk.com/hc/en-us/articles/360002648058-How-do-I-sign-up-for-ag-Grid-release-notifications-). This way you'll know as soon as a new version is out and you can check whether this specific item was implemented on [the changelog page](https://www.ag-grid.com/ag-grid-changelog/).
 
Thanks again for bringing this up with us.

**Comment by dmathisen** [comment]
Thank you!

**Comment by dmathisen** [comment]
Hi @StephenCooper,
Sorry to bother you - I don't see the item **AG-8321** in the [pipeline](https://www.ag-grid.com/pipeline/) or [changelog](https://www.ag-grid.com/changelog/). Maybe I'm misunderstanding, but I wanted to ask and confirm it's in the pipeline somewhere. Thanks for any clarification!

**Comment by StephenCooper** [comment]
Hi, 

I am not 100% sure why its not appearing, likely due to the Jira missing some label which will get resolved. However, more importantly it looks like this bug has already been fixed in our test environment meaning it should be resolved in the next release of 29.3
